flink-python/setup.py

```diff
@@ -224,7 +224,7 @@ def remove_if_exists(file_path):
         author_email='dev@flink.apache.org',
         python_requires='>=3.5',
         install_requires=['py4j==0.10.8.1', 'python-dateutil==2.8.0', 'apache-beam==2.19.0',
-                          'cloudpickle==1.2.2', 'avro-python3>=1.8.1,<=1.9.1'],
+                          'cloudpickle==1.2.2', 'avro-python3>=1.8.1,<=1.9.1', 'jsonpickle==1.2'],
```

Comment:
What about make it an optional requirement such as following?
`
extras_require={
            'ml': ['jsonpickle==1.2']
}
`

Reply:
From the user's perspective and the size of jsonpickle(74.7 kB), maybe keep jsonpicke in install_requires is more user-friendly? i.e., users don't need to specify the ml arguments when installing pyflink.

Reply:
From user's perspective, not every PyFlink user is a ML user and so it doesn't make sense to force a non-ML user to install a dependency which is only used in ML. 

Maybe we should define a consistent stragey for such dependencies. Handling it from the perspective of package size is one option, but I'm not sure if it's the best. For example, what to do with the other dependencies which may be introduced in the future, e.g. pandas for vectorized Python UDF support, numpy for ML support, etc? These dependencies maybe not small. Should we make them as optional or required dependencies?

Reply:
To be honest, make all the dependencies as required seems not bad for me :)

Reply:
Based on a quick look at the python [packaging doc](https://packaging.python.org/discussions/install-requires-vs-requirements/#install-requires-vs-requirements-files) seems like `install_requires` refers to the bare minimum that is needed for it to function. Does `jsonpickle` satisfies that requirement? 

Making all the dependencies as required is one quick solution. but I am afraid that this might lead to issues managing dependencies going forward?

Reply:
Since we only introduce jsonpickle now and the dependency is rather small, it does not hurt non-ML users while brings convenience for ML users. I find more advantages here.
If we add NumPy support for ML, we can add all these dependencies into the optional at that time.